homeassistant/components/influxdb/sensor.py

```diff
@@ -215,7 +281,16 @@ def __init__(self, hass, influx, query):
                 query.get(CONF_IMPORTS),
                 query.get(CONF_GROUP_FUNCTION),
             )
-
+        elif query[CONF_LANGUAGE] == LANGUAGE_RAW:
+            query_clause = query.get(CONF_QUERY)
+            query_clause.hass = hass
+            self.data = InfluxRawSensorData(
+                influx,
+                query.get(CONF_API_VERSION),
+                query.get(CONF_DB_NAME),
+                query_clause,
+                query.get(CONF_FIELD),
+            )
```

Comment:
```suggestion
            if query[CONF_API_VERSION] == DEFAULT_API_VERSION:
                self.data = InfluxV1RawSensorData(
                    influx, query_clause, query.get(CONF_DB_NAME), query.get(CONF_FIELD)
                )
            else:
                self.data = InfluxV2RawSensorData(
                    influx,
                    query_clause,
                )
```

Instead of checking the API version on each update call, we do it once in the init function.